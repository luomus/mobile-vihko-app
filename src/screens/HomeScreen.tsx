import React, { Component } from 'react'
import { Text, View } from 'react-native'
import { Icon } from 'react-native-elements'
import Colors from '../styles/Colors'
import HomeComponent from '../components/home/HomeComponent'
import Cs from '../styles/ContainerStyles'
import Ts from '../styles/TextStyles'
import Bs from '../styles/ButtonStyles'
import { NavigationStackProp, NavigationStackScreenProps } from 'react-navigation-stack'
import { NavigationActions, StackActions } from 'react-navigation'

type Props  = {
  navigation: NavigationStackProp<any, any>
}

export default class HomeScreen extends Component<NavigationStackScreenProps<Props>> {

  static navigationOptions = ({ screenProps, navigation }: any) =>  ({
    title: screenProps.t('mobile vihko'),
    headerStyle: {
      backgroundColor: Colors.headerBackground,
    },
    headerTintColor: Colors.white,
    headerLeft: () => null,
    headerRight: () =>
      <View style={Cs.languageContainer}>
        <Text style={Ts.languageText} onPress={() => {screenProps.i18n.changeLanguage('fi')}}>FI</Text>
        <Text style={Ts.languageText} onPress={() => {screenProps.i18n.changeLanguage('sv')}}>SV</Text>
        <Text style={Ts.languageText} onPress={() => {screenProps.i18n.changeLanguage('en')}}>EN</Text>
        <Icon iconStyle={Bs.headerButton} name='info' type='material-icons' size={25} onPress={() => navigation.navigate('Info')}/>
      </View>
  })

  render() {
    const { dispatch, isFocused, navigate, state } = this.props.navigation
    return (
      <HomeComponent
        isFocused={() => isFocused()}
        onPressMap={() => navigate('Map')}
        onLogout={() => dispatch(
          StackActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'Login' })]
          })
        )}
        onPressObservationEvent={(id: string) => {
          this.props.navigation.navigate('ObservationEvent', { id })
        }}
        obsStopped={state.params?.obsStopped}
        onFinishObservationEvent={() => {
          this.props.navigation.navigate('EditObservationEvent')
        }}
      />
    )
  }
}