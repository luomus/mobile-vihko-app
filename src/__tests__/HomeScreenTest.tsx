import React from 'react'
import HomeScreen from '../screens/HomeScreen'

import renderer from 'react-test-renderer'

describe('<HomeScreen />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })

})
