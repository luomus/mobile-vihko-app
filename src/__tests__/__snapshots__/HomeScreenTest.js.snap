// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "padding": 10,
        "width": "40%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          N채yt채 kartta
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": "40%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Lis채채 havainto
        </Text>
      </View>
    </View>
  </View>
</View>
`;
